spyder/plugins/editor/widgets/tests/conftest.py

```diff
@@ -50,23 +50,82 @@ def codeeditor_factory():
     return editor
 
 
+def editor_factory(new_file=True, text=None):
+    editorstack = EditorStack(None, [])
+    editorstack.set_find_widget(FindReplace(editorstack))
+    editorstack.set_io_actions(Mock(), Mock(), Mock(), Mock())
+    if new_file:
+        if not text:
+            text = ('a = 1\n'
+                    'print(a)\n'
+                    '\n'
+                    'x = 2')  # a newline is added at end
+        finfo = editorstack.new('foo.py', 'utf-8', text)
+        return editorstack, finfo.editor
+    return editorstack, None
+
+
 @pytest.fixture
 def setup_editor(qtbot):
     """
     Set up EditorStack with CodeEditor containing some Python code.
     The cursor is at the empty line below the code.
     Returns tuple with EditorStack and CodeEditor.
     """
-    text = ('a = 1\n'
-            'print(a)\n'
-            '\n'
-            'x = 2')  # a newline is added at end
-    editorStack = EditorStack(None, [])
-    editorStack.set_find_widget(FindReplace(editorStack))
-    editorStack.set_io_actions(Mock(), Mock(), Mock(), Mock())
-    finfo = editorStack.new('foo.py', 'utf-8', text)
-    qtbot.addWidget(editorStack)
-    return editorStack, finfo.editor
+    editorstack, code_editor = editor_factory()
+    qtbot.addWidget(editorstack)
+    return editorstack, code_editor
+
+
+@pytest.fixture
+def completions_editor(
+        completion_plugin_all_started, qtbot_module, request, capsys,
+        tmp_path):
+    """Editorstack instance with LSP services activated."""
+    # Create a CodeEditor instance
+    editorstack, editor = editor_factory(new_file=False)
+    editorstack.set_format_on_save(True)
```

Comment:
Is it possible to move this line to the test that you created below? That way, other tests could reuse this fixture for other purposes.